bdd/spec/027_seq_spec.sh

```diff
@@ -258,11 +258,11 @@ error: sequence out-of-bounds"
         import @std/app
         from @std/seq import seq, Self, recurse
 
-        fn triangularRec(x: int) : int = seq(x + 1).recurse(fn (self: Self, x: int) : Result<int> {
+        fn triangularRec(x: int) : int = seq(x + 1 || 0).recurse(fn (self: Self, x: int) : Result<int> {
           if x == 0 {
             return ok(x);
           } else {
-            return ok(x + (self.recurse(x - 1) || 0));
+            return x + (self.recurse(x - 1 || 0) || 0);
```

Comment:
Same issue here. Type inference inside of recursive functions is a bit broken and we need to fix it. But since we already want to redo this obtuse syntax I don't think this should be a blocker?